### DISCUSSION TITLE
XCode Build never ends with XCode 16

### DISCUSSION THREAD

**Discussion by ajviera** [codemagic-ci-cd/codemagic-docs#2822] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

1. Having an app in Flutter 3.24.3
2. Use the latest version of Xcode and the default version of Cocoapods.
3. Run the building process in Codemagic to create the IPA for distribution.

### Expected results

The building process ends at some point.

### Actual results

The building process is infinitely stuck in “Running Xcode build”.

### Note
If I use XCode 15.4 the build completes without problem, but I can't upload the file to App Store Connect because it is an unsupported version of XCode.

### Build id (optional)

673373c83dfe87efa1092196

**Comment by dam-ease** [comment]
Hello @ajviera,
What Xcode version do you use locally? Also, do you run into any issues when building locally?
Kindly try cleaning your project locally as well using below commands:
```
cd ios
pod cache clean --all
rm -rf ~/Library/Caches/CocoaPods
rm -rf Pods
rm -rf ~/Library/Developer/Xcode/DerivedData/*
pod deintegrate
flutter clean
flutter pub get
pod install
```

**Reply by ajviera** [reply]
Hello @dam-ease 
Locally, I use XCode 16.1 and have no problem building the app in release mode, neither do I nor anyone on my team.

**Reply by dam-ease** [reply]
Hello @ajviera,
Apologies, I missed your last response!
I see your recent builds have been successful, do you still run into this issue after cleaning your project with the commands I shared earlier on?

**Reply by ajviera** [reply]
Hi @dam-ease, I was able to build the app, but using XCode 15.4
If I select any version of XCode 16 in Codemagic the same thing happens. 

Those commands you passed me, I had already used them and I used them again after you told me.
But it doesn't change anything, since the problem is not local. It happens only inside Codemagic

**Reply by dam-ease** [reply]
Trying to clean the project is to confirm this doesn't occur locally.
I have a suggestion you should try out. Please try building with the latest version of cocoapods and Verify that your Podfile uses an appropriate iOS platform version and does not have deprecated settings. Lastly, kindly enable --verbose so that we can get more details about why the build is stuck.
```
flutter build ipa --release --verbose
```
And share the new buildID if the issue still occurs.

**Reply by habyaad** [reply]
@ajviera how did you resolve your issue?

**Reply by dam-ease** [reply]
Hey @habyaad,
Are you running into similar issue now as well?
Please could you try running the build command with the --verbose and share your buildID so we can look into it?

**Reply by habyaad** [reply]
resolved it already, thanks!
